scripts/validate_string_concatenation.py

```diff
@@ -0,0 +1,148 @@
+
+OUTPUT_FORMATS: Dict[str, str] = {
+    "default": "{source_path}:{line_number}:\t between {start} and {end}\n",
+    "azure": (
+        "##vso[task.logissue type=error; {source_path}:{line_number}:\t"
+        "between {start} and {end}\n]"
+    ),
+}
+
+
+def main(*, source_path: str = ".", output_format: str = "default") -> bool:
+    """
+    Main entry point of the script.
+
+    Parameters
+    ----------
+    source_path : str, default '.'
+        Source path representing path to a file/directory.
+    output_format : {'default', 'azure'}, default 'default'
+        Output format of the script.
+
+    Returns
+    -------
+    bool
+        True if found any strings that needs to be concatenated.
+
+    Raises
+    ------
+    ValueError
+        If the `source_path` is not pointing to existing file/directory.
+    """
+    if not os.path.exists(source_path):
+        raise ValueError(
+            "Please enter a valid path, pointing to a valid file/directory."
+        )
+
+    is_failed: bool = False
+    msg: str = OUTPUT_FORMATS.get(output_format)
+
+    if os.path.isfile(source_path):
+        for values in strings_to_concatenate(source_path):
+            is_failed = True
+            print(msg.format(**values))
+
+    for subdir, _, files in os.walk(source_path):
+        for file_name in files:
+            if any(
+                file_name.endswith(extension) for extension in FILE_EXTENSIONS_TO_CHECK
+            ):
+                for values in strings_to_concatenate(os.path.join(subdir, file_name)):
+                    is_failed = True
+                    print(msg.format(**values))
+    return is_failed
+
+
+def strings_to_concatenate(source_path: str) -> Generator[Dict[str, str], None, None]:
+    """
+    Yielding the strings that needs to be concatenated in a given file.
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    Dict[str, str]
+        Containing:
+            source_path
+                Source file path.
+            line_number
+                Line number of unconcatenated string.
+            start
+                Starting string of unconcatenated string.
+            end
+                Ending string of unconcatenated string.
+    """
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for current_token, next_token in zip(tokens, tokens[1:]):
+        if current_token[0] == next_token[0] == token.STRING:
+            yield {
+                "source_path": source_path,
+                "line_number": current_token[2][0],
+                "start": current_token[1],
+                "end": next_token[1],
+            }
+
+
+if __name__ == "__main__":
+    argparser = argparse.ArgumentParser(description="Validate concatenated strings")
+    argparser.add_argument(
+        "--format",
+        "-f",
+        default="default",
```

Comment:
```suggestion
        default="{source_path}:{line_number}:{start}:{end}:{msg}",
```
So by default we use this format, and if the caller ones a different format, can call the script like `script -f "##vso[task.logissue type=error; {source_path}]{msg}" path`. So you don't need to know the constants with the formats beforehand, just the variables.

I'm unsure whether `start` and `end` provide value, are they the columns of the strings to concatenate? I'd say the line number should make clear enough where the problem is, if that's the case.

Reply:
> I'm unsure whether `start` and `end` provide value.

#### Short summary:

```start``` and ```end``` are used to pinpoint the unconcatenated string.

---

#### Long summary:

```start``` and ```end``` are the two unconcatenated strings that needs to be concatenated.

lets say we have a file called ```script.py``` and it's located in ```path/to/script.py``` and on line ```1337``` there is a line of:

```python
print("foo" "bar")
```

When running ```scripts/validate_string_concatenation.py```

The default the output is looking like this:

```
./path/to/script.py:1337      BETWEEN "foo" AND "bar"
```

and the yielded dictionary is mapped like this: 

```python
source_path: "./path/to/script.py"
line_number: 1337
start: "foo"
end: "bar"
```